### PR TITLE
add an alert dialog for swipe up link - fix issue #1101

### PR DESCRIPTION
**Pull request by stamatiap**
If you have any other suggestions for the message that appears in the dialog, please tell me and I can change it.

### CODE REVIEW

File: app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java [hunk 1]
```diff
@@ -442,10 +443,34 @@ public boolean onFling(final MotionEvent e1, final MotionEvent e2, final float v
...
+
+		               	default:
+		               		dialog.dismiss();
+		               		break;
+		           	}
+		      	}
+		  		};
+
+		   		AlertDialog.Builder builder = new AlertDialog.Builder(context);
+		   		builder.setTitle("Are you sure you want to open this link?");
```
Comment (ammargitham) [inline app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java:470]: Extract the string to a string resource

Comment (stamatiap) [reply app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java:470]: Alright, I'll do that right away. But is the message okay? Or should I change it?

File: app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java [hunk 2]
```diff
@@ -443,10 +444,34 @@ public boolean onFling(final MotionEvent e1, final MotionEvent e2, final float v
...
+		               		dialog.dismiss();
+		               		break;
+		           	}
+		      	}
+		  		};
+
+		   		AlertDialog.Builder builder = new AlertDialog.Builder(context);
+		   		builder.setTitle(R.string.swipe_up_confirmation);
+		   		builder.setMessage(tag.toString()).setPositiveButton("Yes", dialogClickListener)
+		   			.setNegativeButton("No", dialogClickListener).show();
```
Comment (austinhuang0131) [inline app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java:473]: You don't need to require the same listener on the negative button. In fact you can just use a null...

Also string resources for buttons (They already exist)

```suggestion
		   		AlertDialog.Builder builder = new AlertDialog.Builder(context);
		   		builder.setTitle(R.string.swipe_up_confirmation);
		   		builder.setMessage(tag.toString()).setPositiveButton(R.string.yes, (d, w) -> Utils.openURL(context, tag.toString()))
		   			.setNegativeButton(R.string.no, null).show();
```

Of course, if you're wary about whether the dialog will actually dismiss or not, you can apply a similar lambda function onto the negative button

**Review by austinhuang0131 — CHANGES_REQUESTED** [review]
We're on new Java, so you can simply shorten the entire function to a lambda instead of importing an entire module and then declare a variable with various functions. Refer to https://github.com/austinhuang0131/barinsta/blob/master/app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java#L220 for a similar dialog implementation

also proper indentations please

File: app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java (hunk 2 above)
Comment (stamatiap) [reply app/src/main/java/awais/instagrabber/fragments/StoryViewerFragment.java:473]: Thank you for the review and the pointers! I'll work on it. 
Also as for the intendations, it looked fine in my IDE, but I guess it changed after I pushed. I'll try to fix that too.

**Review by austinhuang0131 — APPROVED** [review]

**Pull request merged**